core/src/main/java/org/apache/carbondata/core/datastore/row/CarbonRow.java

```diff
@@ -49,6 +49,14 @@ public CarbonRow(Object[] data, Object[] rawData) {
     return data;
   }
 
+  public Object[] getKey(int[] sortColumnIndex) {
+    Object[] key = new Object[sortColumnIndex.length];
```

Comment:
can we reuse the object array ? For each row we are creating the more objects now. How it reduces GC ?

Reply:
I agree with @ajantha-bhat , sortBy will call keyBy first, and keyBy uses map method to generate a (key, value) tuple, with above way, it will copy a key object array for every row. GC time will be reduced in this way? @marchpure, can you test some cases and show performance report ?